Junit test case org.postgresql.test.util.HostSpecTest has incorrect asserts causing failures

**I'm submitting a bug report**

  <!--- What kind of an issue is this? Put an `x` in all the boxes that apply: -->
- [X ] bug report
- [ ] feature request
    

**Describe the issue**
Test Case HostSpecTest , tests testShouldResolveWithEmptySocksProxyHost and testShouldResolveWithWhiteSpaceSocksProxyHost 

are guaranteed to fail. 

Should both use assertTrue,  but they use assertFalse

Note that the tests NEVER runs normally since the super-pom has a surefire include filter for test suites only, and there is no suite in the util package.

N.B. The test and the code tested are very simple.  No database connection required.


**Driver Version ** 
42.2.5

**Java Version?**
1.8

**OS Version?**
n/a

**PostgreSQL Version?**
n/a

**To Reproduce**
Steps to reproduce the behaviour:

mvn    clean package test  -Dtest=HostSpecTest


**Expected behaviour**
A clear and concise description of what you expected to happen.
And what actually happens

Test should pass.
Test fails.

e.g.:
mvn    clean package test  -Dtest=HostSpecTest
...
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.postgresql.test.util.HostSpecTest
Tests run: 8, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.038 sec <<< FAILURE! - in org.postgresql.test.util.HostSpecTest
testShouldResolveWithWhiteSpaceSocksProxyHost(org.postgresql.test.util.HostSpecTest)  Time elapsed: 0.006 sec  <<< FAILURE!
java.lang.AssertionError: null
        at org.junit.Assert.fail(Assert.java:86)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertFalse(Assert.java:64)
        at org.junit.Assert.assertFalse(Assert.java:74)
        at org.postgresql.test.util.HostSpecTest.testShouldResolveWithWhiteSpaceSocksProxyHost(HostSpecTest.java:46)

testShouldResolveWithEmptySocksProxyHost(org.postgresql.test.util.HostSpecTest)  Time elapsed: 0 sec  <<< FAILURE!
java.lang.AssertionError: null
        at org.junit.Assert.fail(Assert.java:86)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertFalse(Assert.java:64)
        at org.junit.Assert.assertFalse(Assert.java:74)
        at org.postgresql.test.util.HostSpecTest.testShouldResolveWithEmptySocksProxyHost(HostSpecTest.java:39)


Results :

Failed tests: 
  HostSpecTest.testShouldResolveWithWhiteSpaceSocksProxyHost:46 null
  HostSpecTest.testShouldResolveWithEmptySocksProxyHost:39 null


**Logs**
If possible PostgreSQL logs surrounding the occurrence of the issue
Additionally logs from the driver can be obtained adding
```java
loggerLevel=TRACE&loggerFile=pgjdbc-trace.log 
```
to the connection string

--- a/pgjdbc/src/test/java/org/postgresql/test/jdbc2/Jdbc2TestSuite.java
+++ b/pgjdbc/src/test/java/org/postgresql/test/jdbc2/Jdbc2TestSuite.java
@@ -18,6 +18,7 @@ import org.postgresql.test.core.JavaVersionTest;
 import org.postgresql.test.core.NativeQueryBindLengthTest;
 import org.postgresql.test.core.OptionsPropertyTest;
 import org.postgresql.test.util.ExpressionPropertiesTest;
+import org.postgresql.test.util.HostSpecTest;
 import org.postgresql.test.util.LruCacheTest;
 import org.postgresql.test.util.ServerVersionParseTest;
 import org.postgresql.test.util.ServerVersionTest;
@@ -45,23 +46,11 @@ import org.junit.runners.Suite;
         ColumnSanitiserDisabledTest.class,
         ColumnSanitiserEnabledTest.class,
         LruCacheTest.class,
+        HostSpecTest.class,
         ReaderInputStreamTest.class,
         ServerVersionParseTest.class,
         ServerVersionTest.class,
 
-        DriverTest.class,
-        ConnectionTest.class,
-        DatabaseMetaDataTest.class,
-        DatabaseMetaDataPropertiesTest.class,
-        SearchPathLookupTest.class,
-        EncodingTest.class,
-        ExpressionPropertiesTest.class,
-        ColumnSanitiserDisabledTest.class,
-        ColumnSanitiserEnabledTest.class,
-        LruCacheTest.class,
-        ReaderInputStreamTest.class,
-        ServerVersionParseTest.class,
-        ServerVersionTest.class,
         OptionsPropertyTest.class,
 
         TypeCacheDLLStressTest.class,

--- a/pgjdbc/src/test/java/org/postgresql/test/util/HostSpecTest.java
+++ b/pgjdbc/src/test/java/org/postgresql/test/util/HostSpecTest.java
@@ -36,14 +36,14 @@ public class HostSpecTest {
   public void testShouldResolveWithEmptySocksProxyHost() throws Exception {
     System.setProperty("socksProxyHost", "");
     HostSpec hostSpec = new HostSpec("localhost", 5432);
-    assertFalse(hostSpec.shouldResolve());
+    assertTrue(hostSpec.shouldResolve());
   }
 
   @Test
   public void testShouldResolveWithWhiteSpaceSocksProxyHost() throws Exception {
     System.setProperty("socksProxyHost", " ");
     HostSpec hostSpec = new HostSpec("localhost", 5432);
-    assertFalse(hostSpec.shouldResolve());
+    assertTrue(hostSpec.shouldResolve());
   }
 
   @Test

